widgets/deployments/src/widget.tsx

```diff
@@ -79,8 +111,10 @@ Stage.defineWidget({
         return obj;
     },
 
-    async fetchData(widget, toolbox, params) {
-        const deploymentDataPromise = new Stage.Common.Deployments.Actions(toolbox.getManager()).doGetDeployments(
+    async fetchData(_widget, toolbox, params): Promise<DeploymentViewData> {
+        const deploymentDataPromise: Promise<PaginatedResponse<Deployment>> = new Stage.Common.Deployments.Actions(
+            toolbox.getManager()
+        ).doGetDeployments(
```

Comment:
Probably more type safe would be to provide proper type out of `doGetDeployments` (I suggest to make it generic in other comment), then I believe that `deploymentDataPromise` will be able to infer the type automatically.

Reply:
I think that infering types would require parsing an _include string, which is too much, made a generic base improvement that allows for some protections

Reply:
Still, it seems to me as not accurate right now. 

You started using `Deployment` type from Deployments View widget in Deployments widget, but we don't fetch the same list of properties from deployment resource. The best would be to have first proper type for Deployment used in Deployments widget. It of course can (and should) base on broader type (the one from Deployments View widget).

In this specific place more accurate would be to have right now:
```suggestion
        const deploymentDataPromise = new Stage.Common.Deployments.Actions(
            toolbox.getManager()
        ).doGetDeployments<
            | 'id'
            | 'display_name'
            | 'blueprint_id'
            | 'visibility'
            | 'created_at'
            | 'created_by'
            | 'updated_at'
            | 'inputs'
            | 'workflows'
            | 'site_name'
            | 'latest_execution'
        >(
```